internal/provider/resource_project_cloud_account_link.go

```diff
@@ -0,0 +1,668 @@
+	}
+
+	return respData.GraphSearch.Nodes[0].Entities[0].ID, nil
+}
+
+func resourceWizProjectCloudAccountLinkCreate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	tflog.Info(ctx, "resourceWizProjectCloudAccountLinkCreate called...")
+	projectID := d.Get("project_id").(string)
+	var cloudAccountWizID string
+
+	if v, ok := d.GetOk("cloud_account_id"); ok {
+		cloudAccountWizID = v.(string)
+	} else {
+		cloudAccountUpstreamID := d.Get("external_cloud_account_id").(string)
+		var diagsSearch diag.Diagnostics
+		cloudAccountWizID, diagsSearch = searchForCloudAccount(ctx, cloudAccountUpstreamID, m)
+		if len(diagsSearch) > 0 {
+			return diagsSearch
+		}
+
+	}
+	// verify that the link does not already exist in wiz
+	// if it does, abort and throw an error, as is standard
+	// terraform behavior (no overwrite or implicit import).
+	partialProject := &PartialProjectWithCloudAccountLinks{}
+	linkExists, requestDiags := checkCloudAccountLinkExistence(ctx, m, projectID, cloudAccountWizID, partialProject)
+	diags = append(diags, requestDiags...)
+	if len(diags) > 0 {
+		return diags
+	}
+
+	if linkExists {
+		return diag.Errorf("cloud account %s is already linked to project %s", cloudAccountWizID, projectID)
+	}
+
+	// link not present, add it to the project
+	newCloudAccountLinksList := make([]*wiz.ProjectCloudAccountLinkInput, len(partialProject.Project.CloudAccountLinks)+1)
+	for i, link := range partialProject.Project.CloudAccountLinks {
+		newCloudAccountLinksList[i] = accountLinkToAccountLinkInput(link)
+	}
+	newCloudAccountLinksList[len(newCloudAccountLinksList)-1] = getAccountLinkVar(d, cloudAccountWizID)
+
+	// define the graphql query for adding the link by taking the existing list and appending
+	// the new entry to it - then patch this property on the wiz project
+	query := `mutation LinkCloudAccountToProject($input: UpdateProjectInput!) {
+		updateProject(input: $input) {
+			project {
+				id
+			}
+		}
+	}`
+
+	// populate the graphql variables
+	vars := &UpdateProjectCloudAccountLinks{
+		ID: projectID,
+		Patch: PatchProjectCloudAccountLinks{
+			CloudAccountLinks: newCloudAccountLinksList,
+		},
+	}
+
+	// process the request
+	data := &UpdateProject{}
+	requestDiags = client.ProcessRequest(ctx, m, vars, data, query, "LinkCloudAccountToProject", "create")
+	diags = append(diags, requestDiags...)
+	if len(diags) > 0 {
+		return diags
+	}
+
+	d.SetId(uuid.NewString())
+	err := d.Set("cloud_account_id", cloudAccountWizID)
+	if err != nil {
+		return append(diags, diag.FromErr(err)...)
+	}
+
+	return resourceWizProjectCloudAccountLinkRead(ctx, d, m)
+}
+
+func resourceWizProjectCloudAccountLinkRead(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	tflog.Info(ctx, "resourceWizProjectCloudAccountLinkRead called...")
+
+	// check the id
+	if d.Id() == "" {
+		return nil
+	}
+
+	// define the graphql query
+	query := `query project  (
```

Comment:
suggest to trim the graph request down to only whats needed for in this context (ids and cloud acc links)